publication-commons/src/main/java/no/unit/nva/doi/models/ImmutableDoi.java

```diff
@@ -0,0 +1,396 @@
+        }
+        return another instanceof ImmutableDoi
+            && equalTo((ImmutableDoi) another);
+    }
+
+    /**
+     * Computes a hash code from attributes: {@code proxy}, {@code prefix}, {@code suffix}.
+     *
+     * @return hashCode value
+     */
+    @Override
+    public int hashCode() {
+        return Objects.hash(proxy, prefix, suffix);
+    }
+
+    /**
+     * Prints the immutable value {@code Doi} with attribute values.
+     *
+     * @return A string representation of the value
+     */
+    @Override
+    public String toString() {
+        return getPrefix() + PATH_SEPARATOR + getSuffix();
+    }
+
+    private static void validateProxyUri(URI proxy) {
+        try {
+            proxy.toURL();
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException(ERROR_PROXY_URI_MUST_BE_A_VALID_URL, e);
+        }
+    }
+
+    private static URI createUriWithoutPathQueryAndFragmentWithSuffixForwardSlash(URI value) {
+        try {
+            return new URI(value.getScheme(), value.getUserInfo(), value.getHost(), value.getPort(),
+                PATH_SEPARATOR_STRING,
+                null, null);
+        } catch (Exception e) {
+            throw new IllegalStateException("Could not reconstruct URI and strip path, query and fragment arguments");
+        }
+    }
+
+    private boolean equalTo(ImmutableDoi another) {
+        return proxy.equals(another.proxy)
+            && prefix.equals(another.prefix)
+            && suffix.equals(another.suffix);
+    }
+
+    /**
+     * Builds instances of type {@link ImmutableDoi ImmutableDoi}. Initialize attributes and then invoke the {@link
+     * #build()} method to create an immutable instance.
+     *
+     * <p><em>{@code Builder} is not thread-safe and generally should not be stored in a field or collection,
+     * but instead used immediately to create instances.</em>
+     */
+    public static final class Builder {
+
+        private static final long INIT_BIT_PREFIX = 0x1L;
+        private static final long INIT_BIT_SUFFIX = 0x2L;
+        private static final long OPT_BIT_PROXY = 0x1L;
+        private long initBits = 0x3L;
+        private long optBits;
+
+        private URI proxy;
+        private String prefix;
+        private String suffix;
+
+        private Builder() {
+        }
+
+        /**
+         * Initializes the value for the {@link Doi#getProxy() proxy} attribute.
+         *
+         * <p><em>If not set, this attribute will have a default value as returned by the initializer of {@link
+         * Doi#getProxy() proxy}.</em>
+         *
+         * @param proxy The value for proxy
+         * @return {@code this} builder for use in a chained invocation
+         */
+
+        public final Builder withProxy(URI proxy) {
+            checkNotIsSet(proxyIsSet(), "proxy");
+            var newProxyValue = createUriWithoutPathQueryAndFragmentWithSuffixForwardSlash(proxy);
+            validateProxyUri(newProxyValue);
```

Comment:
validation should be in one method called `validate()` that would be called in the constructor or build()

Reply:
Fixed and moved into build() on builder.